janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java

```diff
@@ -44,111 +26,109 @@
 import org.apache.tinkerpop.gremlin.structure.Vertex;
 import org.apache.tinkerpop.gremlin.structure.io.graphson.GraphSONMapper;
 import org.apache.tinkerpop.gremlin.structure.io.gryo.GryoMapper;
-import org.apache.velocity.Template;
-import org.apache.velocity.VelocityContext;
-import org.apache.velocity.app.Velocity;
-
-import com.google.common.base.Joiner;
-import com.google.common.collect.ImmutableMap;
 import org.janusgraph.graphdb.tinkerpop.JanusGraphIoRegistry;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
 
-public abstract class AbstractJanusGraphAssemblyIT {
+import java.net.Socket;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
-    protected static final String BUILD_DIR;
-    protected static final String EXPECT_DIR;
-    protected static final String ZIPFILE_PATH;
-    protected static final String ZIPFILE_EXTRACTED;
-    protected static final String ZIPFILE_FULL_PATH;
-    protected static final String ZIPFILE_FULL_EXTRACTED;
+import static org.apache.tinkerpop.gremlin.driver.ser.AbstractMessageSerializer.TOKEN_IO_REGISTRIES;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTimeout;
 
-    static {
-        Properties props;
+public abstract class AbstractJanusGraphAssemblyIT extends JanusGraphAssemblyBaseIT {
 
-        try {
-            props = new Properties();
-            java.io.FileReader fr = new FileReader(Joiner.on(File.separator).join(new String[] { "target", "test-classes", "target.properties" }));
-            props.load(fr);
-            fr.close();
-        } catch (IOException e) {
-            throw new AssertionError(e);
-        }
+    protected abstract String getConfigPath();
+
+    protected abstract String getServerConfigPath();
 
-        BUILD_DIR    = props.getProperty("build.dir");
-        EXPECT_DIR   = props.getProperty("expect.dir");
-        ZIPFILE_PATH = props.getProperty("zipfile.path");
-        ZIPFILE_EXTRACTED = ZIPFILE_PATH.substring(0, ZIPFILE_PATH.length() - 4);
-        ZIPFILE_FULL_PATH = props.getProperty("zipfile-full.path");
-        ZIPFILE_FULL_EXTRACTED = ZIPFILE_FULL_PATH.substring(0, ZIPFILE_FULL_PATH.length() - 4);
+    protected abstract String getGraphName();
 
-        Properties p = new Properties();
-        p.put("file.resource.loader.path", EXPECT_DIR);
-        Velocity.init(p);
+    protected Boolean supportsIndex() {
+        return false;
```

Comment:
I am just thinking... shall we make it as true by default? If we make it true, then if someone adds a subclass in the future which **does not** support index, the test case will immediately fail and they would know that they should override this method. On contrary, let's say we make it false by default, then if someone adds a subclass in the future which **does** support index (without noting they should override this method), index tests will be skipped, which is not ideal.

Reply:
i will check